### ISSUE TITLE
Selecting a community should add a definitionExpression to all community layers

### ISSUE THREAD

**Issue by torrin47** [Innovate-Inc/EnviroAtlas#189]
It looks like a feature layer has a method "setDefinitionExpression" which will filter the layer:  https://developers.arcgis.com/javascript/3/jsapi/featurelayer-amd.html#setdefinitionexpression
all of the community layers appear to have a community attribute - the "Community" field - that could be used for this definitionExpression.  So, when someone selects "Durham, NC", all the communities in the TOC would execute:  featureLayer.setDefinitionExpression("Community = 'Durham, NC'");
Baohong Ji (bji@innovateteam.com)
So it has nothing to do with the filter function in the AttributeTable, right?
hehehehe.  no, probably not. i'm not sure, it's entirely possible that using the filter function in the AttributeTable actually does the same thing, and I'm very curious to see whether the AttributeTable will recognize the DefinitionExpression we set on the layer directly, but because the JavaScript API has that simple method, I think it would be far more straightforward to use it directly than to have one widget call another widget.
Baohong Ji (bji@innovateteam.com)
What is the aim of using the code of "featureLayer.setDefinitionExpression("Community = 'Durham, NC'");"? to change symbology?
well, the thought is that the dynamic symbology widget will recognize the definition expression and include it in calculating statistics.  Esri is usually pretty good about this sort of thing.
Baohong Ji (bji@innovateteam.com)
So what I need to do is just add that piece of code when I add the layer. That is, only show those features when add the layer, right?
well, yes, it will need to apply both to any newly added community layers as well as any community layers already added to the map.
Baohong Ji (bji@innovateteam.com)
What do you mean by "any community layers already added to the map"?
say someone adds a bunch of community layers to the map, then goes and selects Durham, NC, from the community selection widget.  that will zoom them to Durham, NC, and at that point, any community layers already added to the map will need to have that definition expression applied.  if they add a new community layer to the map while Durham is selected, it will also need that definition expression applied.  but then, if they switch to Fresno, CA, the definitionExpression will need to be changed for all those layers.
Baohong Ji (bji@innovateteam.com)
So you mean the all the community layers will be shown for only one community? Is there any case that the user want to add one layer for community Durham, and then he want to add another layer but with community Fresno? I mean can Durham and Fresno coexist?
i think the agreement was no - only one community could be selected at one time for the entire map. if someone wants to compare communities, we will need to implement the multiple map split screen functionality.
Baohong Ji (bji@innovateteam.com)
I have one related question: if the user have already added one community, say Durham. Then he selects another community in "Select Community" widget, Fresno but did not check on any of the layers in search filter. In this case, will the definitionExpression switch to Fresno?
When you say that someone has already added one community, does that mean that they have selected that community in the Select Community Widget?
but they haven't added any map layers?
Baohong Ji (bji@innovateteam.com)
Yes. he had selected Durham in "Select Community widget" and also he added a layer by check on one layer in "search filter". But layer if he open "Select Community widget" and select another community but make no further action (have not check on any selectable layers yet), Should we switch to the latter?
yes.  the layer that was already selected needs to switch to Fresno
Baohong Ji (bji@innovateteam.com)
I see. Are you sure that "the dynamic symbology widget will recognize the definition expression"? If not, why do not we just make the dynamic symbology widget do the calculating statistics?
well, because of how the client described the workflow - that selecting a community would trigger certain actions, including recalculation of the statistics for community layers.  i'm not certain of the recalculation, but if we do it the other way, then the user would have to go through layer by layer to update symbology, rather than having it happen all at once when a community is selected.
Baohong Ji (bji@innovateteam.com)
Ok. I will add the expressionDefintion for the added layers.

**Comment by torrin47** [comment]
So in ticket #179, David discovered the bug that a definition expression that includes a comma in the attribute (Durham, NC) will fail and return all features, which may be why the initial attempt to apply a definition expression via the community selection widget failed on our first attempt. Now that Barb is adding an attribute we can use for this definition expression that doesn't have a comma, let's take a fresh look at setting the expression right up front.

**Comment by Baohong** [comment]
The current map servers (49.16 and production) does not seem to honor definition expression. I retrieved the code when definition expression works (version of Aug 23, 2016) and upload to 49.16. Link is 
https://leb.epa.gov/Projects/EnviroAtlas/CurrentDevelopment/EnviroAtlas_Definition_Expression/index.html
It used to work even at our code meeting before and the feature services are from https://enviroatlas.epa.gov/arcgis/rest/services/Other. But now I cannot see any layers honor expression definition. Have we republished any service on 49.16 and production?
